app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt

```diff
@@ -229,9 +268,92 @@ private fun CancelRequestButton(
               navigateWithToast(navigationActions, context, "Request cancelled successfully")
             })
       },
+      testTag = "cancelRequestButton",
+      icon = {
+        Icon(Icons.Default.Close, contentDescription = null, modifier = Modifier.size(20.dp))
+      },
+      isError = true)
+}
+
+@Composable
+private fun CancelLessonButton(
+    lesson: Lesson,
+    currentProfile: Profile,
+    lessonViewModel: LessonViewModel,
+    navigationActions: NavigationActions,
+    context: Context
+) {
+  var showCancelDialog by remember { mutableStateOf(false) }
+
+  LessonActionButton(
+      text = "Cancel Lesson",
+      onClick = { showCancelDialog = true },
       testTag = "cancelButton",
       icon = {
         Icon(Icons.Default.Close, contentDescription = null, modifier = Modifier.size(20.dp))
       },
       isError = true)
+
+  if (showCancelDialog) {
```

Comment:
Nice add that enhance the user experience but I'm not sure it is the best thing to show the cancel button if it is less than 24h before the lesson because it is useless as the user cannot cancel anyway.

Reply:
I thought that first too, but then I feel like it could be good that the user know why he cannot cancel. But it is a good point and we could talk about it more during next sprint if we want to update the button!